python/lsst/daf/butler/script/queryCollections.py

```diff
@@ -72,37 +80,65 @@ def _getTable(
         names=("Name", typeCol, descriptionCol),
         dtype=(str, str, str),
     )
+    if show_dataset_types:
+        table.add_column(Column(name="Dataset Types", dtype=str))
     butler = Butler.from_config(repo)
+
+    def addCollection(info: CollectionInfo, relative: str) -> None:
+        info_relatives = getattr(info, relative)
+        # Parent results can be returned in a non-deterministic order, so sort
+        # them to make the output deterministic.
+        if relative == "parents":
+            info_relatives = sorted(info_relatives)
+        if info_relatives:
+            collection_table = Table([[info.name], [info.type.name]], names=("Name", typeCol))
+            description_table = Table(names=(descriptionCol,), dtype=(str,))
+            for info_relative in info_relatives:
+                relative_table = Table([[info_relative]], names=(descriptionCol,))
+                if show_dataset_types:
+                    cinfo = butler.collections.get_info(info_relative, include_summary=True)
```

Comment:
Looks like this code is completely untested.

Reply:
Added new unit tests to test `TABLE`, `TREE` and `FLATTEN` (`_getTable`, `_getTree`, `_getList` functions).